src/index.js

```diff
@@ -8,6 +8,37 @@ const ES6_BROWSER_EMPTY = resolve( __dirname, '../src/empty.js' );
 const CONSOLE_WARN = ( ...args ) => console.warn( ...args ); // eslint-disable-line no-console
 const exts = [ '.js', '.json', '.node' ];
 
+let readFileCache = {};
+const readFileAsync = file => new Promise((fulfil, reject) => fs.readFile(file, (err, contents) => err ? reject(err) : fulfil(contents)));
+const statAsync = file => new Promise((fulfil, reject) => fs.stat(file, (err, contents) => err ? reject(err) : fulfil(contents)));
+function cachedReadFile (file, cb) {
+	if (file in readFileCache === false) {
+		readFileCache[file] = readFileAsync(file).catch(err => {
+			delete readFileCache[file];
+			throw err;
+		});
+	}
+	readFileCache[file].then(contents => cb(null, contents), cb);
+}
+
+let isFileCache = {};
+function cachedIsFile (file, cb) {
+	if (file in isFileCache === false) {
+		isFileCache[file] = statAsync(file)
+			.then(
+				stat => stat.isFile(),
+				err => {
+					if (err.code == 'ENOENT') return false;
+					throw err;
+				}
```

Comment:
Why not just put this into the catch block below? Also, using *both* the second `then` argument *and* `catch` seems a little over the top to me... 😉